### DISCUSSION TITLE
Phisics Support

### DISCUSSION THREAD

**Discussion by Omixxx** [motion-canvas/motion-canvas#1112] [Q&A] [answered]
Hi Jacob! great software, I admire your ability to make such a great product while working and developing your own game! I could spend hours asking you how you maintain this level of performance and productivity 😁

Anyway, I'm curious to know if Motion Canvas is able to provide (or integrate) some physics primitives. It would be neat if we could easily apply physics to scenes and objects. 
What do you think? Have you ever tried to do that?

**Comment by aarthificial** [comment, marked as answer]
Hi, thanks for the kind words!

We don't provide physics primitives at the moment (other than springs) but @ksassnowski actually implemented his own [physics engine for Motion Canvas](https://github.com/ksassnowski/motion-canvas-components/tree/main/src/components/physics) if you want to take a look. This is of course a crazy thing to do but it shows that it's possible.

In practice, it should be pretty straight-forward to integrate something like Rapier2D with Motion Canvas but I don't know of any preexisting packages/repos that would provide something out-of-the-box.
